### PR TITLE
4884570: StreamPrintService.isAttributeValueSupported does not work properly for SheetCollate

### PR DESCRIPTION
**Pull request by prrace**
This fixes a longstanding consistency issue where the Postscript StreamPrintService would report SheetCollate.UNCOLLATED in a list of supported values then report it was not supported under the same conditions.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-4884570](https://bugs.openjdk.java.net/browse/JDK-4884570): StreamPrintService.isAttributeValueSupported does not work properly for SheetCollate

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6963/head:pull/6963` \
`$ git checkout pull/6963`

Update a local copy of the PR: \
`$ git checkout pull/6963` \
`$ git pull https://git.openjdk.java.net/jdk pull/6963/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6963`

View PR using the GUI difftool: \
`$ git pr show -t 6963`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6963.diff">https://git.openjdk.java.net/jdk/pull/6963.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/print/attribute/StreamServiceSheetCollateTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        StreamPrintServiceFactory[] fact =
+          StreamPrintServiceFactory.lookupStreamPrintServiceFactories(
+                null, null);
+
+        if (fact.length == 0) {
+            return;
+        }
+        OutputStream out = new ByteArrayOutputStream();
+        StreamPrintService sps = fact[0].getPrintService(out);
+        if (!sps.isAttributeCategorySupported(SheetCollate.class)) {
```
Comment (mrserb) [inline test/jdk/javax/print/attribute/StreamServiceSheetCollateTest.java:52]: Can we add other attributes to the tests? I made a quick check for some attributes and it looks like the code is quite confused, on my windows, the same issue exists for the "OrientationRequested.class".

Comment (prrace) [reply test/jdk/javax/print/attribute/StreamServiceSheetCollateTest.java]: I've also fixed OrientationRequested and added other Enum attributes to the test.

File: src/java.desktop/share/classes/sun/print/PSStreamPrintService.java [hunk 2]
```diff
@@ -440,7 +450,7 @@ else if (attr.getCategory() == Copies.class) {
             if (flavor != null &&
                 !(flavor.equals(DocFlavor.SERVICE_FORMATTED.PAGEABLE) ||
                 flavor.equals(DocFlavor.SERVICE_FORMATTED.PRINTABLE))) {
-                return false;
+                return attr == SheetCollate.UNCOLLATED;
```
Comment (prsadhuk) [inline src/java.desktop/share/classes/sun/print/PSStreamPrintService.java:453]: Shouldn't we need to do the same for SheetCollate in WIn32PrintService#isATtributeValueSupported() where it always return false?

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**